src/topics/inputs.md

```diff
@@ -143,13 +143,56 @@ The `baseCommand` field will always appear in the final command line before the
 
 [touch]: http://www.linfo.org/touch.html
 
+### Optional Input Parameters
+
+Optional input parameters must include `label` and `secondaryFiles`.
+`label` is a short, human-readable description for the parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    label: Use case for label parameter
+    inputBinding:
+      position: 1
+```
+
+`secondaryFiles` is an optional input parameter that provides a pattern of specifying files
+or directories that must be included alongside the primary file. 
+The following example demonstrates the `secondaryFiles` input parameter.
+
+```cwl
+inputs:
+  example_file:
+    type: File
+    secondaryFiles: [example_file.txt]
+```
+
+Also, a file object listed in `secondaryFiles` may contain nested `secondaryFiles` as shown below:
```

Comment:
You're confusing "secondaryFiles" on the input parameter definitions, and "secondaryFiles" appearing in a parameter File value.

When it appears in the input parameter definition, it is describing "these are the secondary files this parameter is expected to have".

When it appears in a File object, it means "these are the secondary files this parameter actually has".

The 1st case (parameter definition) can't be nested.

The 2nd case (File object) it can be nested.

It's important to understand that the parameter definition is used to _check_ the input object, so if you have these input values:

```
{
  class: File
  location: primary.txt
  secondaryFiles: [
    {
      class: File,
      location: primary.txt.idx
    }
  ]
}
```

The corresponding parameter definition is something like this:

```
inputs:
  example_file:
    type: File
    secondaryFiles: [.idx]
```